### DISCUSSION TITLE
Feature request - scheduled task on first of month

### DISCUSSION THREAD

**Discussion by AChundee** [KelvinTegelaar/CIPP#2331] [Ideas]
Hi All,

It would be great if I could schedule tasks to happen on the first of the month (even better if I could also schedule them to happen on the last day of the month too!). My specific use-case is to run a licence overview to assist with billing.

Apologies if someone else has asked for this already, or it's possible already and I've missed how to do this. TIA.

**Comment by bandicootlimited** [comment]
Seconded. I have been trying for an hour to get a list of Identity Management \ Administration \ Users \ All Tenants and it always times out. Be great to get this delivered on Monday morning or 1st of every month so I can report to clients.

**Comment by KelvinTegelaar** [comment]
this isn't the place for feature requests, but thankfully this already exists. You can use the scheduler to set this up. Two methods:

- Go to CIPP -> Scheduler, schedule "Get-cipplicenseoverview" to run a license overview report as you see it in CIPP, for example in our HaloPSA integration this would look like: 
![image](https://github.com/user-attachments/assets/29faa17f-e411-4416-8eee-8faa90d3f7ec)

- Go to CIPP -> Tenant Administration -> Tools -> Graph Explorer. You can make any custom report here.
